ui/src/app/tags/components/TagsHeader/TagsHeader.test.tsx

```diff
@@ -0,0 +1,26 @@
+import { render, screen } from "@testing-library/react";
+import { Provider } from "react-redux";
+import { MemoryRouter } from "react-router-dom";
+import configureStore from "redux-mock-store";
+
+import TagsHeader from "./TagsHeader";
+
+import { TagHeaderViews } from "app/tags/constants";
+import { rootState as rootStateFactory } from "testing/factories";
+
+const mockStore = configureStore();
+
+it("can display the add tag form", () => {
+  const store = mockStore(rootStateFactory());
+  render(
+    <Provider store={store}>
+      <MemoryRouter initialEntries={[{ pathname: "/pools", key: "testKey" }]}>
+        <TagsHeader
+          headerContent={{ view: TagHeaderViews.AddTag }}
+          setHeaderContent={jest.fn()}
+        />
+      </MemoryRouter>
+    </Provider>
+  );
+  expect(screen.getByText("Add tag form")).toBeInTheDocument();
```

Comment:
```suggestion
  expect(screen.getByRole("form", { name: "Add tag" })).toBeInTheDocument();
```

Reply:
Could you use the appropriate role. This can easily get missed when you open another PR with the full form implementation.

Reply:
I think it's OK, that text isn't going to stay...

Reply:
I've added a todo here.